presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -73,8 +74,16 @@ public QueryStateInfoResource(
         return userQueries.stream()
                 .map(queryInfo -> createQueryStateInfo(
                         queryInfo,
-                        queryManager.getQueryResourceGroup(queryInfo.getQueryId()),
-                        queryManager.getQueryResourceGroup(queryInfo.getQueryId()).map(ResourceGroupId::getRoot).map(rootResourceGroupInfos::get)))
+                        getQueryResourceGroup(queryInfo),
+                        getQueryResourceGroup(queryInfo).map(ResourceGroupId::getRoot).map(rootResourceGroupInfos::get)))
                 .collect(toImmutableList());
     }
+
+    private Optional<ResourceGroupId> getQueryResourceGroup(QueryInfo queryInfo) {
+        try {
+            return queryManager.getQueryResourceGroup(queryInfo.getQueryId());
```

Comment:
This is a weird API. It throws if the query doesn't exist, but returns an optional resource group (since the query might not have one). And the only caller (this one), doesn't care about that distinction.

So, let's make it return optional for either case and never throw.